cpp/include/cuspatial/cubic_spline.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/table/table.hpp>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute quadratic interpolations of a set of points based on their
+ * ids and a coefficient matrix.
+ *
+ * @param[in] query_points float32 column of coordinate values to be 
+ * interpolated.
+ *
+ * @param[in] curve_ids int32 column of ids associated with each query point.
+ * These ids identify which spline each query point should be interpolated
+ * into.
+ *
+ * @param[in] prefixes int32 column of prefix_sum of the source_points.
```

Comment:
I think you should call this offsets.

```suggestion
 * @param[in] offsets column of offsets to the start of the source_points for each spline.
```

Reply:
Thanks for all your writing suggestions! I have implemented almost all of them, including renaming prefixes to offsets throughout.